lib/matplotlib/tests/test_legend.py

```diff
@@ -74,6 +74,56 @@ def test_labels_first():
     ax.legend(loc=0, markerfirst=False)
 
 
+@image_comparison(baseline_images=['legend_fig_noargs'], extensions=['png'],
```

Comment:
Is there a reasonable way to check that this works without an image test?

Reply:
Happy to try that.  Or reduce the tests.  I don’t know how you would test it without a plot but that’s just my ignorance.  Happy to implement something

Reply:
You only need to test equivalence between the results of different invocations, not equivalence to some human-verified gold standard output.

Reply:
see https://github.com/matplotlib/matplotlib/pull/9305/files#diff-663c17d0729218d60dbb862c286a17d4R5327

Reply:
OK< I think I understand.  Use the trick linked above and then assert that the pngs are the same w/o actually having to save them or do the image comparison?  That sounds great.  

Sorry for being slow, but I don't know *anything* about writing tests for python...

Reply:
And I'm not an expert at testing with matplotlib, but I would've thought there might be a way to check the calculated legend artists are equivalent. But perhaps not.

Reply:
`ax.legend_.get_texts()` and `ax.legend_.legendHandles`  _seems_ to get you back the labels and handles used, but that is just from poking at a legend object interactively....